SonarRunner.Shim/PropertiesFileGenerator.cs

```diff
@@ -298,7 +303,40 @@ private static void AddOrSetProperty(string key, string value, AnalysisPropertie
             }
         }
 
+        /// <summary>
+        /// Passing in an invalid value for the build wrapper output directory will cause the C++ plugin to
+        /// fail (invalid = missing or empty directory) so we'll remove invalid settings
+        /// </summary>
+        private static void RemoveBuildWrapperSettingIfDirectoryEmpty(AnalysisProperties properties, ILogger logger)
```

Comment:
This code is inside the PropertiesFileGenerator class. It's not ideal having plugin-specific code inside this class but there isn't a clearly better alternative. Also, there is already special-case property handling code in this class for the obsolete VSBootstrapper property key, so there is a precedent.